django/db/models/sql/compiler.py

```diff
@@ -531,7 +531,6 @@ def get_combinator_sql(self, combinator, all):
         compilers = [
             query.get_compiler(self.using, self.connection, self.elide_empty)
             for query in self.query.combined_queries
-            if not query.is_empty()
```

Comment:
This is premature; empty queries will raise `EmptyResultSet` anyway and be handled below in the `except EmptyResultSet` branch.

Making sure `compilers` are not filtered here ensures that when the _sliced union_ branch_ is reached below (`if self.query.is_sliced and combinator == "union"`) then `len(compilers) > 1` and thus the `compiler.elide_empty = False` assignment ensures at least two `SELECT` get compiled and thus the singleton `LIMIT` over `LIMIT` issue is adverted.